cst438_gradebook/src/main/java/com/cst438/controllers/GradeBookController.java

```diff
@@ -152,6 +155,74 @@ public void updateGradebook (@RequestBody GradebookDTO gradebook, @PathVariable(
 		
 	}
 	
+	//change an assignment's name
+	@PutMapping("/gradebook/{id}/{newName}")
+	@Transactional
+	public void updateAssignmentName (@PathVariable("id") Integer assignmentId, @PathVariable("newName") String newName ) {
+		
+		String email = "dwisneski@csumb.edu";  // user name (should be instructor's email) 
+		Assignment assignment = checkAssignment(assignmentId, email);  // check that user name matches instructor email of the course.
+		
+		assignment.setName(newName);
+		assignmentRepository.save(assignment);
+		
+		
+	}
+	
+	// delete an assignment
+	@DeleteMapping("/gradebook/{id}")
+	@Transactional
+	public void deleteAssignment(  @PathVariable("id") Integer assignmentId  ) {
+		
+		String email = "dwisneski@csumb.edu";   // instructor's email 
+		
+		Assignment assignment = checkAssignment(assignmentId, email);
+		
+		//reject deletion if assignment is graded
+		if (assignment.getNeedsGrading() == 1) {
+			throw  new ResponseStatusException( HttpStatus.BAD_REQUEST, "Assignments with existing grades cannot be deleted.");
+		}
+		
+		// get the enrollment for the course
+		//  for each student, if an (empty) grade file exists, such as those created by get gradebook/{id}, delete it
+		GradebookDTO gradebook = new GradebookDTO();
+		gradebook.assignmentId= assignmentId;
+		gradebook.assignmentName = assignment.getName();
+		for (Enrollment e : assignment.getCourse().getEnrollments()) {
+			GradebookDTO.Grade grade = new GradebookDTO.Grade();
+			grade.name = e.getStudentName();
+			grade.email = e.getStudentEmail();
+			// does student have a grade for this assignment
+			AssignmentGrade ag = assignmentGradeRepository.findByAssignmentIdAndStudentEmail(assignmentId,  grade.email);
```

Comment:
Might be helpful to have a more descriptive name here than ag for code readability, and might be nice to have this super long line indented onto the next line.